src/cli/rustup_mode.rs

```diff
@@ -404,6 +404,12 @@ pub fn cli() -> App<'static, 'static> {
                     SubCommand::with_name("uninstall")
                         .about("Uninstall a toolchain")
                         .alias("remove")
+                        .arg(
+                            Arg::with_name("regex")
+                                .help("Use a regular expression to select the toolchains to uninstall")
+                                .long("regex")
+                                .takes_value(false),
```

Comment:
I did not add the option of `-r` since that's often used to mean "recursive" (e.g. `grep -r`) and I don't want to confuse people.

Reply:
I'm wondering if we need this at all (flag that is) -- instead we could just treat the name as a pattern if there's a `*` in it.